pkgs/development/compilers/dmd/default.nix

```diff
@@ -47,9 +126,11 @@ stdenv.mkDerivation rec {
 
       cd ../phobos
       mkdir $out/lib
-      ${let bits = if stdenv.is64bit then "64" else "32";
-            osname = if stdenv.isDarwin then "osx" else "linux"; in
-      "cp generated/${osname}/release/${bits}/libphobos2.a $out/lib"
+      ${
+          let bits = builtins.toString stdenv.hostPlatform.parsed.cpu.bits;
+          osname = if stdenv.isDarwin then "osx" else "linux";
```

Comment:
Now also fixed, thanks again.

One question. While working on those two derivations I sometimes made changes to them but `nix-build -K -A dmd` did not trigger a rebuild.
I was under the impression that changing the derivation file in any way should always lead to a new hash and therefore need to trigger a rebuild.
I then thought that maybe nix is expanding the script in any form and can therefore cancel out some changes without a rebuild.
Any way, this happend again with this last commit and I simply added two empty new lines somewhere to trigger a rebuild.